### PR TITLE
Implement .gptignore

### LINKED ISSUE #5: Implement .gptignore
In `autopr.utils.repo.repo_to_file_descriptors`, make it read a file named `.gptignore` in the root of the repo, and filter according to that. Allow for star syntax, like in `.gitignore`.

Also, make a new file named `.gptignore`, containing `*.lock`.

### PR DESCRIPTION
**Pull request by garrettApproachableGeek**
Implements the `.gptignore` functionality as described in issue https://github.com/irgolic/AutoPR/issues/5

Alternative to https://github.com/irgolic/AutoPR/pull/6

Fixes #5

### CODE REVIEW

File: autopr/utils/repo.py [hunk 1]
```diff
@@ -107,6 +112,8 @@ def repo_to_file_descriptors(repo: Repo, context_window: int, file_chunk_size: i
         if blob.type == 'tree':
             continue
         try:
+            if is_path_ignored(blob.path, ignore_patterns):
+                continue
```
Comment (irgolic) [inline autopr/utils/repo.py:116]: Can you move this out of the `try` block?

File: autopr/utils/repo.py [hunk 2]
```diff
@@ -139,3 +146,29 @@ def repo_to_file_descriptors(repo: Repo, context_window: int, file_chunk_size: i
...
+def parse_gptignore(repo: Repo) -> list[str]:
+    gptignore_file = ".gptignore"
+    ignore_patterns = []
+
+    try:
+        gptignore_blob = repo.head.commit.tree / gptignore_file
+        gptignore_content = gptignore_blob.data_stream.read().decode()
+    except FileNotFoundError:
+        return ignore_patterns
+
```
Comment (irgolic) [inline autopr/utils/repo.py:167]: I thought this threw a `KeyError`, not `FileNotFoundError` on failure :o

But regardless, maybe let's get rid of the `try` block anyway, and let it raise all the way through on an error reading the content.
```suggestion
def parse_gptignore(repo: Repo, gptignore_filename: str = ".gptignore") -> list[str]:
    tree = repo.head.commit.tree
    if gptignore_filename not in tree:
        return []
    gptignore_blob = tree / gptignore_filename
    gptignore_content = gptignore_blob.data_stream.read().decode()
    
    ignore_patterns = []
```

**Review by irgolic — CHANGES_REQUESTED** [review]

File: autopr/utils/repo.py (hunk 1 above)
Comment (garrettApproachableGeek) [reply autopr/utils/repo.py]: Moved above the `try` block

File: autopr/utils/repo.py (hunk 2 above)
Comment (garrettApproachableGeek) [reply autopr/utils/repo.py]: Removed `try` block

**Review by irgolic — APPROVED** [review]

**Comment by irgolic** [comment]
Nice, thank you!

**Pull request merged**